commit d9408716d2126439fbc46f6c40e72792069b8411
Author: Mario Limonciello <mario.limonciello@amd.com>
Date:   Fri Jun 23 08:49:56 2023 -0500

    crypto: ccp - Add support for setting user ID for dynamic boost control
    
    As part of the authentication flow for Dynamic Boost Control, the calling
    software will need to send a uid used in all of its future
    communications.
    
    Add support for another IOCTL call to let userspace software set this up.
    
    Acked-by: Tom Lendacky <thomas.lendacky@amd.com>
    Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/drivers/crypto/ccp/dbc.h b/drivers/crypto/ccp/dbc.h
index 1c3a0a078d15..156435100076 100644
--- a/drivers/crypto/ccp/dbc.h
+++ b/drivers/crypto/ccp/dbc.h
@@ -33,9 +33,15 @@ struct dbc_nonce {
 	struct dbc_user_nonce		user;
 } __packed;
 
+struct dbc_set_uid {
+	struct psp_req_buffer_hdr	header;
+	struct dbc_user_setuid		user;
+} __packed;
+
 union dbc_buffer {
 	struct psp_request		req;
 	struct dbc_nonce		dbc_nonce;
+	struct dbc_set_uid		dbc_set_uid;
 };
 
 void dbc_dev_destroy(struct psp_device *psp);